### DISCUSSION TITLE
Can I bind to the value of a select?

### DISCUSSION THREAD

**Discussion by isaacfink** [huntabyte/shadcn-svelte#301] [Help]
How can I bind to the value of a select? I tried this
```
<SelectInput name="name" bind:value={info.timezone} />
```
but the value does not get updated, is there any way to do it reactively or only with a callback?

**Comment by isaacfink** [comment]
If it's possible to bind my next question would be how to ensure type safety, for example if `timezone` is a string how can I be sure every option value will be a string, it gets more complicated when dealing with multiple

**Comment by owensco** [comment]
Same question. My workaround is to use an event dispatcher every time the select changes (`onSelectedChange`), as mentioned.

**Comment by dhndmn** [comment]
I'm also looking for help here. I'm wanting to set a default selected value from the database if it exists so the user doesn't have to reselect each time. I've tried using each of the Select properties from Bits UI and Melt UI on the Select.Root and Select.Item components with no luck.

In Bits UI it looks like this should work:
`<Select.Root selected={fieldname}>`

Melt's API uses 'defaultSelected'

**Reply by dhndmn** [reply]
Update: got my issue working by combining some suggestions at https://github.com/huntabyte/shadcn-svelte/issues/361

Needed to use an object with value AND label.
